server/testutils/containere2e/containere2e.go

```diff
@@ -14,7 +17,25 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+var buildPluginOnce sync.Once
+
+func buildPlugin(t *testing.T) {
+	cmd := exec.Command("make", "-C", "../../", "dist")
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "DEFAULT_GOOS=linux")
+	cmd.Env = append(cmd.Env, "DEFAULT_GOARCH=amd64")
```

Comment:
Are we sure that the test runners are going to be `amd64`? I guess it can end up being `arm64` too.

Reply:
You mention about the docker containers, but I guess docker containers in arm processors should use arm architecture, I think we should be building for the current runner architecture and the GOOS linux, but I'm not 100% sure how it works in Mac.

Reply:
I'm thinking now that maybe, we can do the plugin compilation here, already inside a testcontainer. Build the container for go, share the filesystem of the current directory, execute the build, and copy over the file outside the container, tear down the container, and use the compiled version. Not needed in this PR, I think we can merge this and think about adding that later.

Reply:
For context, the issue here is matching the architecture of the Mattermost enterprise docker containers -- otherwise the plugin won't run.

Once we start building and supporting Mattermost for arm, we'll probably need to revisit this.

Reply:
Good point, you are 100% right.